### PR TITLE
update docs with docker toolbox & machine details

### PR DESCRIPTION
**Pull request by bbodenmiller**
Cleanup docs to reference Docker Toolbox and Machine where appropriate rather than depreciated `boot2docker` CLI tool.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,27 +67,12 @@ The following example will enable core dumps inside containers, but you can
 specify any other options you may need.
 
 ```console
-boot2docker ssh -t sudo vi /var/lib/boot2docker/profile
+docker-machine default ssh -t sudo vi /var/lib/boot2docker/profile
```
Comment (yosifkit) [inline README.md]: Swap order of `ssh` and `default`.

File: README.md [hunk 2]
```diff
@@ -129,27 +67,12 @@ The following example will enable core dumps inside containers, but you can
...
 # Add something like:
 #     EXTRA_ARGS="--default-ulimit core=-1"
-boot2docker restart
+docker-machine
```
Comment (yosifkit) [inline README.md]: Probably still wanted restart here?  `docker-machine restart default`

File: README.md [hunk 3]
```diff
@@ -167,7 +90,7 @@ $ docker run -v /data --name my-data busybox true
 $ # Share it using Samba (Windows file sharing)
 $ docker run --rm -v /usr/local/bin/docker:/docker -v /var/run/docker.sock:/docker.sock svendowideit/samba my-data
 $ # then find out the IP address of your Boot2Docker host
-$ boot2docker ip
+$ docker-machine ip
```
Comment (yosifkit) [inline README.md]: This still needs `default`, like all the docker-machine commands :cry:

File: README.md [hunk 4]
```diff
@@ -242,17 +165,15 @@ As of Docker version 1.3.1, if your registry doesn't support HTTPS, you must add
...
+$ docker-machine ssh default "echo $'EXTRA_ARGS=\"--insecure-registry <YOUR INSECURE HOST>\"' | sudo tee -a /var/lib/boot2docker/profile && sudo /etc/init.d/docker restart"
 ```
 
 then you should be able to do a docker push/pull.
 
 #### Running behind a VPN (Cisco AnyConnect, etc)
 
 So sometimes if you are behind a VPN, you'll get an `i/o timeout` error.
-The current work around is to forward the port in the boot2docker-vm.
+The current work around is to forward the port in the `boot2docker-vm`.
```
Comment (yosifkit) [inline README.md]: `boot2docker-vm` might get confused as a command.

File: README.md [hunk 5]
```diff
@@ -301,15 +214,15 @@ pass: tcuser
 Boot2docker uses [Tiny Core Linux](http://tinycorelinux.net), which runs from
 RAM and so does not persist filesystem changes by default.
 
-When you run `boot2docker init`, the `boot2docker` tool auto-creates a disk that
+When you run `docker-machine`, the tool auto-creates a disk that
 will be automounted and used to persist your docker data in `/var/lib/docker`
 and `/var/lib/boot2docker`.  This virtual disk will be removed when you run
-`boot2docker delete`.  It will also persist the SSH keys of the machine.
+`docker-machine delete`.  It will also persist the SSH keys of the machine.
```
Comment (yosifkit) [inline README.md]: `default` again :wink:

**Comment by yosifkit** [comment]
Thanks for taking this on!  I left a few comments, if you have some more time.  Thanks again.

**Comment by bbodenmiller** [comment]
Addressed your comments. I believe technically `default` can be dropped from the `docker-machine` commands but it still demonstrates how you can target certain machines.

**Comment by yosifkit** [comment]
LGTM, cc @tianon

**Comment by tianon** [comment]
Very nice, thanks! :+1:

LGTM

**Pull request merged**